pringles/simulator/simulation.py

```diff
@@ -0,0 +1,232 @@
+                           engine='python',  # C engine doesnt work for regex
+                           converters=df_converters,
+                           names=[cls.TIME_COL, cls.PORT_COL, cls.VALUE_COL])
+
+    @classmethod
+    def _parse_main_log_file(cls, file_path):
+        log_file_per_component = {}
+        parsed_logs = {}
+        with open(file_path, 'r') as main_log_file:
+            main_log_file.readline()  # Ignore first line
+            log_dir = os.path.dirname(file_path)
+            for line in main_log_file:
+                name, path = line.strip().split(' : ')
+                log_file_per_component[name] = path if os.path.isabs(path) else log_dir + '/' + path
+
+        df_converters = {
+            cls.VALUE_COL: cls._parse_value,
+            cls.TIME_COL: VirtualTime.parse
+        }
+        for logname, filename in log_file_per_component.items():
+            parsed_logs[logname] = pd.read_csv(filename,
+                                               delimiter=r' /\s+',
+                                               engine='python',  # C engine doesnt work for regex
+                                               converters=df_converters,
+                                               names=[0, 1,  # Not sure what first two cols are
+                                                      cls.MESSAGE_TYPE_COL,
+                                                      cls.TIME_COL,
+                                                      cls.MODEL_ORIGIN_COL,
+                                                      cls.PORT_COL,
+                                                      cls.VALUE_COL,
+                                                      cls.MODEL_DEST_COL])
+        return parsed_logs
+
+    def plot_port(self, logname: str, portname: str,
+                  axes: Optional[Axes] = None, index=0) -> Optional[Axes]:
+        log: pd.DataFrame = self.logs_dfs[logname]
+        data_to_plot = log[log[self.PORT_COL] == portname]
+        if data_to_plot.empty:
+            return None
+        if isinstance(data_to_plot[self.VALUE_COL][0], tuple):
+            y_values = data_to_plot[self.VALUE_COL].map(lambda x: x[index])
+        else:
+            y_values = data_to_plot[self.VALUE_COL]
+
+        if axes is None:
+            axes = plt.axes()  # Create a new axes in current figure
+        x_values = data_to_plot[self.TIME_COL]
+        axes.plot(x_values, y_values)
+        x_values_labels = [str(vtime) for vtime in sorted(x_values, key=float)]
+        axes.set_xticklabels(x_values_labels, rotation=90)
+
+        axes.set_xlim(left=0)
+        return axes
+
+    def get_process_output(self) -> str:
+        """Returns the process STDOUT converted to string
+
+        :return: the process STDOUT
+        :rtype: str
+        """
+        return self.process_result.stdout.decode("utf-8")
+
+
+class Simulation:
+
+    DEFAULT_PICKLEFILE_NAME = 'simulation.pkl'
+
+    def __init__(self,
```

Comment:
Please add docstring with params description. This will be the most used method in our API, so it would be nice to at least have it here.